### PR TITLE
[ARCTIC-875] Avoid printing duplicate error logs

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

related pr: https://github.com/NetEase/arctic/pull/872

## Brief change log

Avoid re-throwing exceptions periodically

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by zhoujinsong — COMMENTED** [review]
@XBaith  I left some comments, PTAL.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java [hunk 1]
```diff
@@ -326,9 +328,16 @@ public void addNewTables(List<TableIdentifier> toAddTables) {
...
         newTableItem.getTableOptimizeRuntime().setOptimizeStatusStartTime(createTime);
         addTableIntoCache(newTableItem, arcticTable.properties(), true);
+        // remove recover table if it is present
+        invalidTables.remove(toAddTable);
         success++;
       } catch (Throwable t) {
-        LOG.error("failed to load  " + toAddTable, t);
+        // avoid print too many error logs
+        if (!invalidTables.contains(toAddTable)) {
+          LOG.error("failed to load  " + toAddTable, t);
```
[the remark this answers is not in the document]
Comment (XBaith) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java:337]: Correct. Because without this condition, the error logs will print over and over again

[the remark this answers is not in the document]
Comment (zhoujinsong) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java:337]: So you may want to move line 339 after line 337?

[the remark this answers is not in the document]
Comment (XBaith) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java:337]: thanks for reminding
